hudi-common/src/main/java/org/apache/hudi/common/model/DefaultHoodieRecordPayload.java

```diff
@@ -149,7 +150,7 @@ protected boolean needUpdatingPersistedRecord(IndexedRecord currentValue,
      * NOTE: Deletes sent via EmptyHoodieRecordPayload and/or Delete operation type do not hit this code path
      * and need to be dealt with separately.
      */
-    String orderField = properties.getProperty(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY);
+    String orderField = ConfigUtils.getOrderingField(properties);
```

Comment:
Could you add tests for these two changes, i.e., if the payload ordering field is not set but the pre-combine field is set?  Check `TestDefaultHoodieRecordPayload` class for how to write a test.